ContentFiles/ElementPackages/StandardLibrary/Path.ele

```diff
@@ -0,0 +1,158 @@
+constraint PathFunc(u:Num):Vector3
+struct Bounds(lower:Num, upper:Num)
+
+
+struct Path(at:PathFunc, domain:Bounds)
+{
+    # A Path is defined by a parametric equation for a path,
+    # and a domain over which the function is valid.
+
+    constraint ScalarTransform(x:Num):Num
+    constraint Vector3Unary(v:Vector3):Vector3
+    constraint Vector3Binary(v1:Vector3, v2:Vector3):Vector3
+
+    changeDomain(this:Path, newDomain:Bounds):Path
+    {
+        # Change the path domain from the current domain "newDomain".
+        #
+        # This function returns a new path "newPath" such that
+        #
+        #   newPath.domain = newDomain
+        #   newPath.at(newDomain.lower) = this.at(this.domain.lower)
+        #   newPath.at(newDomain.upper) = this.at(this.domain.lower)
+
+        uTransform(u:Num):Num
+        {
+            return = Num.lerp(u.sub(newDomain.lower).div(newDomain.upper.sub(newDomain.lower)), this.domain.lower, this.domain.upper)
+        }
+        transformedPathFunc(u:Num):Vector3 = this.at(uTransform(u))
+        return = Path(transformedPathFunc, newDomain)
+    }
+
+    pingPong(this:Path):Path
+    {
+        # Returns a path which pingPongs along the original path
+        # in the original domain.
+        #
+        # Eg, if the original domain is (0, 1), then the returned
+        # path also has a domain of (0, 1) has pingPonged from end to end.
```

Comment:
`path also has a domain of (0, 1) has` sounds a bit weird. `path also has a domain of (0, 1) which has` ?

Reply:
Good spot 👀  Will rephrase